Feature/frictionless payments

lounge opened the pull request.

lounge committed:
Refactor address-related classes and improve payment order handling

Refactored classes related to addresses by moving them under 'PaymentOrder' namespace for better organization. Introduced 'Payer', 'RiskIndicatorDto', and related DTO classes to handle payment order requests more efficiently. Also updated the 'PaymentOrderDto' to incorporate these newly introduced classes. A series of new settings have been added to the '.DotSettings' file as part of this update.

lounge committed:
Refactor address handling and enrich Payer details

The address handling has been refactored by unifying the BillingAddress and PickUpAddress into a single Address class. Several new fields have been added to the Payer class to fill out the payer's detailed information. This includes fields such as FirstName, LastName, and PayerReference along with indicators for Account Age, Password Changes, Suspicious Activity, and more. The RiskIndicatorDto and PayerDto classes have also been updated to make some properties nullable, reflecting the optional nature of these fields in the payment process.

zunkas commented on src/SwedbankPay.Sdk.Infrastructure/PaymentOrder/Address/AddressDto.cs:
Flytta ut till rooten

zunkas commented on src/SwedbankPay.Sdk.Infrastructure/PaymentOrder/Payer/PayerDto.cs:
Döp om till PayerRequestDto

zunkas commented on src/SwedbankPay.Sdk.Infrastructure/PaymentOrder/Payer/PayerResponse.cs:
```diff
@@ -5,9 +5,19 @@ namespace SwedbankPay.Sdk.Infrastructure.PaymentOrder.Payer;
         Device = dto.Device?.Map();
+        Reference = dto.Reference;
+        Name = dto.Name;
+        Email = new EmailAddress(dto.Email ?? "");
```
Kolla null or whitespace på emal och telefon, och gör nytt objekt om den inte är null bara

zunkas commented on src/SwedbankPay.Sdk.Infrastructure/RiskIndicatorDto.cs:
Döp om till RiskIndicatorRequestDto och flytta in till egen mapp

zunkas commented on src/SwedbankPay.Sdk/Msisdn.cs:
Uppdatera Msisdn på Details object till att använda denna class istället för string

zunkas commented on src/SwedbankPay.Sdk/PaymentOrder/Address/Address.cs:
Flytta ut till root

zunkas commented on src/SwedbankPay.Sdk.Infrastructure/PaymentOrder/Payer/PayerDto.cs:
```diff
@@ -0,0 +1,37 @@
+    
+    public AddressDto? BillingAddress { get; }
+    
+    public AddressDto? ShippingAddress { get; }
```
Skicka med AccountInfo

zunkas commented on src/SwedbankPay.Sdk/RiskIndicator.cs:
Flytta in till egen mapp

lounge committed:
Add AccountInfo to Payer and refactor namespaces

Added the AccountInfoDto object to the Payer object to provide additional payer details. Refactored several namespaces for better organization of the codebase. Now, classes like Msisdn, Address, RiskIndicator have been correctly placed. This should improve the readability and maintainability of the project.

zunkas committed:
Merge branch 'develop' into feature/frictionless-payments

zunkas committed:
Update Msisdn and manage related functionalities

This commit changes the Msisdn value in CheckOutController.cs and adds a getter in Msisdn.cs. The changes also involve updating the related test codes, including the Msisdn value usage in the swish-e2e.spec.cy.js, and commenting out a piece of code that attempts to type the Msisdn value into an input field in swedbank-block.ts.
